helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -2061,4 +2070,37 @@ public ZKHelixAdmin build() {
               _zkAddress), false);
     }
   }
+
+  private Map<String, InstanceConfig> findTimeoutOfflineInstances(String clusterName,
+      long offlineDuration) {
+    Map<String, InstanceConfig> instanceConfigMap = new HashMap<>();
+    // in case there is no customized timeout value, use the one defined in cluster config
+    if (offlineDuration == VALUE_NOT_SET) {
+      offlineDuration =
+          _configAccessor.getClusterConfig(clusterName).getOfflineNodeTimeOutForPurge();
+      if (offlineDuration == VALUE_NOT_SET) {
+        return instanceConfigMap;
+      }
+    }
+
+    HelixDataAccessor accessor =
+        new ZKHelixDataAccessor(clusterName, new ZkBaseDataAccessor<ZNRecord>(_zkClient));
```

Comment:
could you double-check if this `zkclient` is a `RealmAwareZkClient`?

Reply:
Yeah, confirmed it's a RealmAwareZkClient.